### PR TITLE
Manage numeric categories

### LINKED ISSUE #219: Update graphs to have numeric categories
Allow to configure numeric categories on graphs

### PR DESCRIPTION
**Pull request by Alteoris**
close #219

### CODE REVIEW

File: OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/CategoryModel.cs [hunk 1]
```diff
@@ -7,6 +7,11 @@ public class CategoryModel
         /// </summary>
         public string Name { get; set; }
 
+        /// <summary>
+        /// Value category name
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/CategoryModel.cs:11]: Category name with a double ?

File: OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/CategoryTypes.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿namespace OpenXMLSDK.Engine.ReportEngine.DataContext.Charts
+{
+    /// <summary>
+    /// Define the category values type
+    /// </summary>
+    public enum CategoryTypes
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/ReportEngine/DataContext/Charts/CategoryTypes.cs:6]: Please use singular name as convention

File: OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartCategory.cs [hunk 3]
```diff
@@ -13,12 +7,17 @@ public class ChartCategory : BaseElement
         /// </summary>
         public string Name { get; set; }
 
+        /// <summary>
+        /// Value category name
+        /// </summary>
+        public double? Value { get; set; }
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/Word/ReportEngine/Models/Charts/ChartCategory.cs:13]: Same comment about the name and double

File: OpenXMLSDK.Engine/Word/ReportEngine/Renders/BarChartExtensions.cs [hunk 4]
```diff
@@ -364,6 +361,43 @@ private static Run CreateBarGraph(BarModel chartModel, OpenXmlPart documentPart)
...
+        /// <summary>
+        /// Create categories
+        /// </summary>
+        /// <param name="chartModel"></param>
+        /// <param name="barChartSeries"></param>
+        private static void ManageCategories(BarModel chartModel, BarChartSeries barChartSeries)
+        {
+            uint p = 0;
+
+            switch (chartModel.CategoryType)
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/Word/ReportEngine/Renders/BarChartExtensions.cs:373]: You should add a default case

File: OpenXMLSDK.Engine/Word/ReportEngine/Renders/LineChartExtensions.cs [hunk 5]
```diff
@@ -195,6 +190,8 @@ public static void ManageLineChart(LineModel chartModel, PlotArea plotArea, UInt
                     catAxis.AppendChild(new CrossesAt() { Val = new DoubleValue(chartModel.CategoriesAxisModel.CrossesAt) });
                 else
                     catAxis.AppendChild(new Crosses() { Val = new EnumValue<CrossesValues>(CrossesValues.AutoZero) });
+                if (chartModel.CategoryType.Equals(CategoryTypes.NumberReference))
+                    catAxis.AppendChild(new DC.NumberingFormat { FormatCode = "#,##0.00", SourceLinked = false });
```
Comment (HisCodeness) [inline OpenXMLSDK.Engine/Word/ReportEngine/Renders/LineChartExtensions.cs:194]: Is it relevant to create a constant for this format also used un bar chart ?

**Review by DamienSAVOYAT — APPROVED** [review]
Ok, for me.

**Pull request merged**